internal/devconfig/configfile/packages.go

```diff
@@ -366,3 +368,134 @@ func packagesFromLegacyList(packages []string) []Package {
 	}
 	return packagesList
 }
+
+// PackageSpec specifies a Devbox package to install. Devbox supports a number
+// of package spec syntaxes:
+//
+//	| Syntax           | Example                     | Description                          |
+//	| ---------------- | --------------------------- | ------------------------------------ |
+//	| name@version     | go@v1.22                    | Resolved w/ search service           |
+//	| name             | go                          | Same as above; implies @latest       |
```

Comment:
This is not really valid. (or should not be). I was under impression that anything without a version is understood to mean:

`nixpkgs/<hash>#attr`

When someone goes `devbox add go` it adds `go@latest` to devbox.json.

If someone does `devbox add stdenv.cc.cc.lib` which is not in the index, then it should add `stdenv.cc.cc.lib` which represents `nixpkgs/<hash>#stdenv.cc.cc.lib`

If someone manually edits the devbox.json to add attribute paths without `@version` flag they get `nixpkgs/<hash>#attr`

Reply:
We also support nixpkgs attribute path via `nixpkgs#hello`

Reply:
Oh wow, I didn't realize that `devbox add go` and manually adding `go` to `devbox.json` did different things. I never would've guessed that `{"packages": ["go"]}` gives me an old version of Go, especially because there's no `nixpkgs.commit` field.

Could we align the CLI and config to both follow the CLI's behavior of checking for an @latest version? Having the same package string mean different things depending on if it's given on the command line vs. a config is pretty confusing IMO.

Reply:
I wonder how often that `@latest` user is accidentally mentioned on GitHub.